features/pages/flaw_detail_page.py

```diff
@@ -239,7 +257,60 @@ def check_value_exist(self, value):
             EC.visibility_of(e)
         )
 
+    def check_value_not_exist(self, value):
+        return WebDriverWait(self.driver, self.timeout).until(
+            EC.invisibility_of_element_located((By.XPATH, f'//span[contains(text(), "{value}")]'))
+        )
+
     def get_text_value(self, field):
         field_value = getattr(self, field + 'Text')
         self.driver.execute_script("arguments[0].scrollIntoView(true);", field_value)
         return field_value.get_text()
+
+    def delete_all_reference(self):
+        # edit ((//div[@class='osim-list-edit'])[n]/div[2]/button)[1]
+        # delete ((//div[@class='osim-list-edit'])[n]/div[2]/button)[2]
+        try:
+            reference_list = find_elements_in_page_factory(self, "referenceList")
+        except NoSuchElementException:
+            # reference not exist, no-op
+            pass
+        else:
+            # delete all reference
+            reference_count = len(reference_list)
+            for i in range(1, reference_count+1):
+                del_btn_xpath = f"((//div[@class='osim-list-edit'])[{i}]/div[2]/button)[2]"
+                del_btn = self.driver.find_element(By.XPATH, del_btn_xpath)
+                self.driver.execute_script("arguments[0].scrollIntoView(true);", del_btn)
+                self.driver.execute_script("arguments[0].click();", del_btn)
+                self.referenceDelConfirmBtn.click_button()
+                self.wait_msg("referenceDeletedMsg")
+                self.toastMsgCloseBtn.click_button()
+
+    def click_add_reference_btn(self):
+        self.driver.execute_script("arguments[0].scrollIntoView(true);", self.addReferenceBtn)
+        self.driver.execute_script("arguments[0].click();", self.addReferenceBtn)
+
+    def click_save_reference_btn(self):
```

Comment:
Is it possible to use one function with line 290? The others LGTM.

Reply:
Yes, i will write a common function to do this.